GitUI/CommandsDialogs/SettingsDialog/Pages/GitConfigSettingsPage.cs

```diff
@@ -133,118 +137,87 @@ protected override void PageToSettings()
-            {
-                exeFile = mergeToolPath;
-                exeName = Path.GetFileName(exeFile);
-            }
-            else
-            {
-                exeFile = MergeToolsHelper.FindMergeToolFullPath(ConfigFileSettingsSet, _NO_TRANSLATE_GlobalMergeTool.Text, out exeName);
-            }
+            var diffMergeToolConfig = _diffMergeToolConfigurationManager.LoadDiffMergeToolConfig(toolName, DifftoolPath.Text);
+            DifftoolCmd.Text = diffMergeToolConfig.FullDiffCommand;
+        }
 
-            if (string.IsNullOrEmpty(exeFile))
+        private void SuggestMergeToolCommand()
+        {
+            var toolName = _NO_TRANSLATE_GlobalMergeTool.Text;
+            if (string.IsNullOrWhiteSpace(toolName))
             {
-                MergetoolPath.SelectAll();
-                MergetoolPath.SelectedText = "";
-                MergeToolCmd.SelectAll();
-                MergeToolCmd.SelectedText = "";
-                if (sender != null)
-                {
-                    MessageBox.Show(this, string.Format(_toolSuggestPathText.Text, exeName),
-                        _mergeToolSuggestCaption.Text);
-                }
-
+                MergeToolCmd.Text = string.Empty;
                 return;
             }
 
-            MergetoolPath.SelectAll(); // allow Undo action
-            MergetoolPath.SelectedText = exeFile;
-            MergeToolCmd.SelectAll();
-            MergeToolCmd.SelectedText = MergeToolsHelper.MergeToolcmdSuggest(_NO_TRANSLATE_GlobalMergeTool.Text, exeFile);
+            var diffMergeToolConfig = _diffMergeToolConfigurationManager.LoadDiffMergeToolConfig(toolName, MergetoolPath.Text);
+            MergeToolCmd.Text = diffMergeToolConfig.FullMergeCommand;
         }
 
+        private void MergeToolCmdSuggest_Click(object sender, EventArgs e)
+            => SuggestMergeToolCommand();
+
         private void DiffToolCmdSuggest_Click(object sender, EventArgs e)
+            => SuggestDiffToolCommand();
+
+        private void DiffMergeToolPath_TextChanged(object sender, EventArgs e)
         {
-            if (!EnvUtils.RunningOnWindows())
+            if (IsLoadingSettings)
             {
                 return;
             }
 
-            var diffToolPath = DifftoolPath.Text.Trim().Trim('"', '\'');
-
-            CurrentSettings.SetPathValue(string.Format("difftool.{0}.path", _NO_TRANSLATE_GlobalDiffTool.Text.Trim()), diffToolPath ?? "");
-            string exeName;
-            string exeFile;
-            if (!string.IsNullOrEmpty(diffToolPath))
-            {
-                exeFile = diffToolPath;
-                exeName = Path.GetFileName(exeFile);
-            }
-            else
+            if (sender == DifftoolPath)
             {
-                exeFile = MergeToolsHelper.FindDiffToolFullPath(ConfigFileSettingsSet, _NO_TRANSLATE_GlobalDiffTool.Text, out exeName);
+                DifftoolPath.Text = DifftoolPath.Text.ToPosixPath();
```

Comment:
If you type a \, the cursor is moved back to first position

Reply:
Sorry, I am unable to repro, perhaps I'm not doing it like you are
Please provide more info

Reply:
Type a path, when entering a '\' the cursor moves to first position
Why not use LostFocus instead on OnChange?

Reply:
Fixed, thank you